docs/proposal/20250224-rosa-networking.md

```diff
@@ -0,0 +1,221 @@
+
+## Summary
+
+This proposal defines implementation of networking infrastructure in CAPA for ROSA Hosted Control Plane.
+
+## Motivation
+
+To be able to provision a new ROSA HCP kubernetes cluster using CAPA, one has to create and setup the underlying network infrastructure first: VPC, public and private subnets, internet gateway, routing tables for both subnets, elastic IP address.
+
+All of the above can be currently provisioned and configured via AWS CLI, AWS Management Console or Terraform. Motivation for this work is to be able to provision and configure the network infrastructure for ROSA HCP using CAPI.
+
+### Goals
+
+1. Implement a namespaced new custom resource `RosaNetwork` representing the networking stack for ROSA HCP.
+2. It will be possible to reference the new `RosaNetwork` resource from ROSA control plane resource
+3. Implement creation and deletion for the new `RosaNetwork` resource.
+4. Support the same networking scenarios as [ROSA CLI](https://github.com/openshift/rosa) using the same embeded AWS CloudFormation template that ROSA CLI uses.
+
+### Non-Goals/Future Work
+
+- Modify current networking code in AWS / EKS clusters.
+- Support custom CloudFormation template.
+
+## Proposal
+
+The goal of this proposal is to be able to provision the networking infrastructure required for a ROSA HCP cluster.
+
+[ROSA CLI](https://github.com/openshift/rosa) supports creation of the networking infrastructure for ROSA HCP and uses [AWS CloudFormation](https://aws.amazon.com/cloudformation/) template under the hood. The [CloudFormation template used by ROSA CLI](https://github.com/openshift/rosa/blob/master/cmd/create/network/templates/rosa-quickstart-default-vpc/cloudformation.yaml) allows to specify five parameters: CloudFormation stack name, AZ count or list of availability zones, region and CIDR block for the VPC. The created CloudFormation stack then contains a VPC, public and private subnets (each pair created in separate AZ), internet gateway attached to VPC, elastic IPs, NAT gateways, public and private routing tables and a security group.
+
+Adopting the CloudFormation template used by rosa-cli would mean that CAPA and the `RosaNetwork` custom resource would be relying on a mechanism that is know to work well and any changes or fixes implemented in ROSA CLI would be picked up automatically in CAPA.
+
+In practical terms, implementation of the proposal would mean:
+1. A new namespaced custom resource definition `RosaNetwork` in CAPA with five attributes: name, AZ count, list of availability zones, region and CIDR block for VPC. `availabilityZoneCount`, `availabilityZones`, `region` and `cidrBlock` will become the `spec` part of the new `RosaNetwork` type, name of the cloudformation stack will be the same as `metadata.name`.
+
+   `RosaNetwork` spec example:
+    ```
+    kind: RosaNetwork
+    metadata:
+      name: rosa-network-01
+      namespace: default
+    spec:
+      availabilityZoneCount: 3
```

Comment:
since we have the availabilityZones, no need for availabilityZoneCount,  what do you think ?

Reply:
I think that as long the CF template we're using supports both scenarios, we can support both as well.

Reply:
Are the `availabilityZones` and `availibilityZoneCount` fields intended to be mutually exclusive?

Based on the two examples provided, I'm assuming they are, however the proposal should state this explicitly.

Reply:
@nrb Those 2 parameters are mutually exclusive in the sense that explicit AZs take priority over the AZ count. But the CF template we'll be using is written so that if you specify both, the stack provisioning succeeds (with explicit AZs taking priority). I can mention this in this in the proposal doc as well.